crates/kubelet/src/provider/mod.rs

```diff
@@ -127,6 +127,11 @@ pub trait Provider: Sized + Send + Sync + 'static {
         Err(NotImplementedError.into())
     }
 
+    /// Gets the path at which to construct temporary directories for volumes.
```

Comment:
@bacongobbler I don't really like this pattern (also used before for `plugin_registry`). It would be too easy to forget to implement these methods, and probably super hard to debug given that there are only two places that make use of the method on the provider, and many that use the method on the state. IMO it is caused by volume functionality being _partially_ tied to `GenericProviderState`. I think it might make more sense for these methods to exist in only one place (Provider state), and for all providers, not just ones implementing `GenericProvider`. This could be enforced with a trait bound and provide the same default implementation that we see here. Thoughts?

Reply:
I agree. Satisfying a Trait instead of overriding a function seems like the more idiomatic approach. Assuming there's a way to determine if a Provider satisfies a given trait, that would also give us a clean abstraction to determine if certain features are supported or not.